### ISSUE TITLE
Investigate compatibility with Python 3.11

### ISSUE THREAD

**Issue by mdickinson** [enthought/scimath#163]
We need to check whether scimath is compatible with Python 3.11:

- Create a Python 3.11 venv
- Install scimath into it (try both from a recent GitHub clone, and from PyPI)
- Run the test suite

If all looks good on Python 3.11, it would be great to have a GitHub Actions workflow that runs the scimath test suite under Python 3.11. If not, we need to open issues and fixes for whatever turns up.

**Comment by homosapien-lcy** [comment]
scimath is tested with:

# setup and activate the virtual env
python3 -m venv ~/python311_test
source ~/python311_test/bin/activate
# install scimath
pip3 install scimath
python3 -m unittest discover scimath

![Screen Shot 2023-03-06 at 1.48.13 PM.png](https://images.zenhubusercontent.com/638fd27786b34c3edb3fdedf/e2598e01-0e52-40fe-ba13-b0c324b3fbca)

All tests have passed. There are two exceptions raised during the unittests, which looks like part of the tests.
I would conclude that scimath is still compatible with python 3.11.

**Comment by mdickinson** [comment]
@homosapien-lcy Thank you! Next steps are:

- Extend the GitHub Actions workflow to test on Python 3.11 as well as Python 3.6 and Python 3.8.
- Please could you open an issue for the unexpected test output? I agree that it looks as though it's intended, but we should be able to find a way to silence the output in the test itself. Ideally we'd have a clean test run.

**Comment by mdickinson** [comment]
> * Extend the GitHub Actions workflow to test on Python 3.11 as well as Python 3.6 and Python 3.8.

Note: this will require some reworking of the workflow to use a version of Python provided by GitHub Actions (via the `setup-python` action) instead of from EDM. It might be cleanest to create a _new_ workflow that runs the SciMath test suite outside of the context of EDM, and doesn't use EDM at all.

Some other ETS packages already have such a workflow: examples are `apptools`, `envisage` and `traits`.

**Comment by homosapien-lcy** [comment]
Sure, will look into it

**Comment by homosapien-lcy** [comment]
@mdickinson Is it something like this in test-with-edm.yml? (just 3.11 instead?) ![Screen Shot 2023-03-15 at 6.03.43 PM.png](https://images.zenhubusercontent.com/638fd27786b34c3edb3fdedf/3a437434-8287-44fa-96aa-15bb5a66c750)

**Comment by mdickinson** [comment]
@homosapien-lcy Yes, something along those lines.

By the way, it's not good practice to post screenshots of code - it obstructs search, and can make it difficult to follow what's going on on some devices. In this case, you can link to the relevant code instead. (See the GitHub documentation for how to do that if you need to.)

**Comment by mdickinson** [comment]
@homosapien-lcy Actually, what we're looking for is something more like this: https://github.com/enthought/envisage/blob/main/.github/workflows/test-with-pypi.yml

Note that in this workflow, we're not using EDM at all: we're testing directly using the GitHub Actions-provided Python versions. It would be great if you could put something together along these lines.

**Comment by mdickinson** [comment]
@homosapien-lcy Just a reminder from above:

> Please could you open an issue for the unexpected test output?

**Comment by homosapien-lcy** [comment]
> The version '3.6' with architecture 'x64' was not found for Ubuntu 22.04."

Thanks, I just added it to issue #170. I also opened a PR #169 to demonstrate the error. Is this a good way to show the unexpected behavior?

**Comment by mdickinson** [comment]
@homosapien-lcy Thanks for opening #170. I think it's fine to not support Python 3.6 for the new workflow.

**Comment by mdickinson** [comment]
> @homosapien-lcy Just a reminder from above:
> 
> > Please could you open an issue for the unexpected test output?

I've opened #171 for this.

**Comment by mdickinson** [comment]
@homosapien-lcy The discussion gets a bit confusing when you make substantial edits to a previous comment, as with https://github.com/enthought/scimath/issues/163#issuecomment-1469615073. In the original comment, you mentioned some unexpected test output, but the edited comment removes all mention of that, which makes the subsequent discussion a bit hard to follow. If there's new information, I'd recommend making a new comment with that new information instead.

**Comment by homosapien-lcy** [comment]
Thanks for the advice Marks! In the future I will avoid too much editting... Also, I found there might be an easy fix for issue #171, I mentioned that in the issue, do you want to take a look?
